src/Helpers/SingleFieldTrait.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace GingerPluginSdk\Helpers;
+
+use GingerPluginSdk\Bases\BaseField;
+use GingerPluginSdk\Exceptions\OutOfDiapasonException;
+use GingerPluginSdk\Interfaces\ValidateFieldsInterface;
+use JetBrains\PhpStorm\Pure;
+
+trait SingleFieldTrait
+{
+    /**
+     * @param $property_name - This attribute will be used while parsing in toArray method as key
+     * @param $value - This attribute will be used while parsing in toArray method as value
+     * @return \GingerPluginSdk\Bases\BaseField
+     */
+    protected function createSimpleField($property_name, $value): BaseField
+    {
+        $new_class = new class($property_name) extends BaseField {
+            #[Pure] public function __construct($property_name)
+            {
+                parent::__construct($property_name);
+            }
+        };
+        $new_class->set($value);
+        return $new_class;
+    }
+
+    protected function createEnumeratedField($property_name, $value, $enum): ValidateFieldsInterface|BaseField
+    {
+        $new_class = new class($property_name, $enum, $value) extends BaseField implements ValidateFieldsInterface {
```

Comment:
The provide $value argument is never used

Reply:
The abstract class seems to make this code more complex then required. Personally I would not use separate validator classes at all. This part can be replaced by an in_array call.

Reply:
The code can certainly be made much more understandable in terms of what we have already seen.

I was trying to make code with specific ideas and messages that could be followed when scaling the library in the future. Since we have a lot of options for what to add to the library, I wanted to create mechanisms at the lowest level that could be easily extended, but not modified.

I used classes to maintain a single interface for all fields so that they can be viewed from the same angle and do the same operations with them without fear that something will go wrong.

In code, sometimes fields such as Country or Date of Birth are implemented with standard classes because anonymous ones look really complicated when heavily loaded in them. At the same time, the small standard classes look lonely and empty, so I replaced them with anonymous ones, for fields like Description or Last Name. The golden mean for me was enumeration fields, as they implement additional validation and at the same time are very similar to each other.

I need separate validation methods to separate workflows. So validation calls are always guaranteed for objects that implement Validation, and I can be sure that all of them are executed and will be executed in the future when scaling the library and adding new entities. Protection from ignorance.

Reply:
>> The provide $value argument is never used
Right, thanks for remark